src/Databases/MySQL/tryQuoteUnrecognizedTokens.cpp

```diff
@@ -0,0 +1,91 @@
+#include <Databases/MySQL/tryQuoteUnrecognizedTokens.h>
+#include <Parsers/CommonParsers.h>
+#include <Common/quoteString.h>
+
+namespace DB
+{
+
+/// Checks if there are no any tokens (like whitespaces) between current and previous pos
+static bool noWhitespaces(const char * to, const char * from)
+{
+    return static_cast<size_t>(from - to) == 0;
+}
+
+/// Checks if the token should be quoted too together with unrecognized
+static bool isWordOrNumber(TokenType type)
+{
+    return type == TokenType::BareWord || type == TokenType::Number;
+}
+
+static void quoteLiteral(
+    IParser::Pos & pos,
+    IParser::Pos & pos_prev,
+    const char *& pos_unrecognized,
+    const char *& copy_from,
+    String & rewritten_query)
+{
+    /// Copy also whitespaces if any
+    const auto * end =
+        isWordOrNumber(pos->type) && noWhitespaces(pos_prev->end, pos->begin)
+        ? pos->end
+        : pos_prev->end;
+    String literal(pos_unrecognized, static_cast<size_t>(end - pos_unrecognized));
+    rewritten_query.append(copy_from, pos_unrecognized - copy_from).append(backQuoteMySQL(literal));
+    copy_from = end;
+    pos_unrecognized = nullptr;
```

Comment:
I'd move `pos_unrecognized = nullptr;` outside of this method - right after calling `quoteLiteral` in line 57, because code below will look more understandable (at first I did not see that `pos_unrecognized` is made `nullptr` here so code below seemed wrong, so seems better to move it from here for readability).

Reply:
it will look like

        if (pos_unrecognized && commit)
        {
            quoteLiteral(
                pos,
                pos_prev,
                pos_unrecognized,
                copy_from,
                rewritten_query);
            pos_unrecognized = nullptr;
        }
....

    if (pos_unrecognized)
    {
        quoteLiteral(
            pos,
            pos_prev,
            pos_unrecognized,
            copy_from,
            rewritten_query);
        pos_unrecognized = nullptr;
    }

Reply:
yeah that's what I meant

Reply:
it is just not obvious when you are reading the code that quoteLiteral will make pos_unrecognized=nullptr, so not quite easy to read this